### PR TITLE
Add division modal

### PR DESCRIPTION
**Pull request by humbertodiazp**
https://github.com/cascaritaco/cascarita/assets/97567622/178f02f2-a195-487b-bcb9-66f5e6dc81da

## Summary by CodeRabbit

- **New Features**
    - Added a new function `createDivision` for creating divisions.
    - Modified the parameter name `group_id` to `groupId` in `createTeam` function.
    - Added `DivisionResponse` to the import statement.

- **Enhancements**
    - Updated the visual style of modal titles to improve readability and aesthetics.

- **Refactor**
    - Modified parameter naming for better consistency in codebase.

### CODE REVIEW

File: client/api/service.tsx [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
+    throw error;
+  }
+};
+
+const createDivision = async (
+  groupId: string,
+  name: string
+): Promise<DivisionResponse> => {
+  try {
+    const response = await fetch(`/api/division/`, {
```
Comment (abanuelo) [inline client/api/service.tsx:28]: nit: remove the `/` and the end

File: client/components/Modal/Modal.module.css [hunk 2]
```diff
@@ -81,8 +81,9 @@
 .title {
   margin-bottom: 16px;
   font-weight: 500;
-  color: #1e1e1e;
-  font-size: 1.5rem;
+  color: #5a5656;
+  font-size: 1.6rem;
```
Comment (abanuelo) [inline client/components/Modal/Modal.module.css:85]: just food for thought, but I know we use `px`, should we use `rem` standards?

**Review by abanuelo — CHANGES_REQUESTED** [review]
Overall looks great! Also great work on the CSS, I can tell you know your stuff. I learned a-lot thank you!

File: client/api/service.tsx (hunk 1 above)
Comment (humbertodiazp) [reply client/api/service.tsx:28]: ack!

File: client/components/Modal/Modal.module.css (hunk 2 above)
Comment (humbertodiazp) [reply client/components/Modal/Modal.module.css:85]: Good question! px is purposely designed to be a fixed size that doesn't scale and rem is designed to scale. You should use both in your layouts depending on whether you want something to scale with the text or not.

**Comment by jipatino98** [comment]
Looks good Humberto! Other than the issues you resolved with Armando, everything looks good!

**Comment by abanuelo** [comment]
Just fix some of the CI builds @humbertodiazp and we should be good to merge!

**Pull request closed without merging**